# Investgenics

Investors need to decide which of the three options (gold, oil, or stocks) is the most stable entity to invest in.

In one of the previous extended study activities, you set up your private GitHub repo and added the data to your system. 
For this module, you’ll have the opportunity to apply your Pandas knowledge to understand your data better and 
solve specific problems for Investgenics. You'll" try to answer the following questions for the firm:

Which stocks were sold with the highest volume?  
Which stocks were sold with the lowest volume?

In [58]:
# Import pandas library 
import pandas as pd

In [62]:
# Create the gold stock price dataframe by reading and loading gold_stocks_price.csv. gsp = gold stock price
gsp = pd.read_csv(r'C:\Users\bocco\OneDrive\Documents\Education\20240708-20250208 - LSE - Data Analytics Career Accelerator'
           r'\Course 2\Module 2\Data\gold_stocks_price.csv')  # gsp = gold stock price

print(gsp.columns)
print(gsp.shape)
print(gsp.dtypes)
gsp.head()

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Adj Close', 'Volume',
       'SP_open', 'SP_high', 'SP_low', 'SP_close', 'SP_Ajclose', 'SP_volume',
       'DJ_open', 'DJ_high', 'DJ_low', 'DJ_close', 'DJ_Ajclose', 'DJ_volume',
       'EG_open', 'EG_high', 'EG_low', 'EG_close', 'EG_Ajclose', 'EG_volume',
       'EU_Price', 'EU_open', 'EU_high', 'EU_low', 'EU_Trend', 'OF_Price',
       'OF_Open', 'OF_High', 'OF_Low', 'OF_Volume', 'OF_Trend', 'OS_Price',
       'OS_Open', 'OS_High', 'OS_Low', 'OS_Trend', 'SF_Price', 'SF_Open',
       'SF_High', 'SF_Low', 'SF_Volume', 'SF_Trend', 'USB_Price', 'USB_Open',
       'USB_High', 'USB_Low', 'USB_Trend', 'PLT_Price', 'PLT_Open', 'PLT_High',
       'PLT_Low', 'PLT_Trend', 'PLD_Price', 'PLD_Open', 'PLD_High', 'PLD_Low',
       'PLD_Trend', 'RHO_PRICE', 'USDI_Price', 'USDI_Open', 'USDI_High',
       'USDI_Low', 'USDI_Volume', 'USDI_Trend', 'GDX_Open', 'GDX_High',
       'GDX_Low', 'GDX_Close', 'GDX_Adj Close', 'GDX_Volume', 'USO_Open',
       'USO_Hig

Date        Open        High         Low       Close   Adj Close  \
0  2011-12-15  154.740005  154.949997  151.710007  152.330002  152.330002   
1  2011-12-16  154.309998  155.369995  153.899994  155.229996  155.229996   
2  2011-12-19  155.479996  155.860001  154.360001  154.869995  154.869995   
3  2011-12-20  156.820007  157.429993  156.580002  156.979996  156.979996   
4  2011-12-21  156.979996  157.529999  156.130005  157.160004  157.160004   

     Volume     SP_open     SP_high      SP_low  ...    GDX_Low  GDX_Close  \
0  21521900  123.029999  123.199997  121.989998  ...  51.570000  51.680000   
1  18124300  122.230003  122.949997  121.300003  ...  52.040001  52.680000   
2  12547200  122.059998  122.320000  120.029999  ...  51.029999  51.169998   
3   9136300  122.180000  124.139999  120.370003  ...  52.369999  52.990002   
4  11996100  123.930000  124.360001  122.750000  ...  52.419998  52.959999   

   GDX_Adj Close  GDX_Volume   USO_Open   USO_High    USO_Low  USO_Close  \
0      48.973877    20605600  36.900002  36.939999  36.049999  36.130001   
1      49.921513    16285400  36.180000  36.500000  35.730000  36.270000   
2      48.490578    15120200  36.389999  36.450001  35.930000  36.200001   
3      50.215282    11644900  37.299999  37.610001  37.220001  37.560001   
4      50.186852     8724300  37.669998  38.240002  37.520000  38.110001   

   USO_Adj Close  USO_Volume  
0      36.130001    12616700  
1      36.270000    12578800  
2      36.200001     7418200  
3      37.560001    10041600  
4      38.110001    10728000  

[5 rows x 81 columns]

In [64]:
# Create the oil price dataframe by reading and loading oil_price.csv. op = oil price
op = pd.read_csv(r'C:\Users\bocco\OneDrive\Documents\Education\20240708-20250208 - LSE - Data Analytics Career Accelerator'
           r'\Course 2\Module 2\Data\oil_price.csv')

print(op.columns)
print(op.shape)
print(op.dtypes)
op.head()

Index(['Date', 'Price', 'Open', 'High', 'Low', 'Vol.'], dtype='object')
(1844, 6)
Date      object
Price    float64
Open     float64
High     float64
Low      float64
Vol.      object
dtype: object


Date   Price    Open    High     Low     Vol.
0  2011-12-01  100.20  100.51  101.17   98.87  275.86K
1  2011-12-02  100.96  100.00  101.56   99.76  245.92K
2  2011-12-05  100.99  101.23  102.44  100.24  239.38K
3  2011-12-06  101.28  100.45  101.42  100.20  217.03K
4  2011-12-07  100.49  101.11  101.94   99.67  330.53K

In [141]:
# Create a subset of the gsp dataframe with only the 'Date', 'Open', 'High', 'Low' columns

gsp_subset = gsp[['Date', 'Open', 'High', 'Low']]
print(gsp_subset.columns)
print(gsp_subset.shape)
print(gsp_subset.dtypes)
gsp_subset.head()

Index(['Date', 'Open', 'High', 'Low'], dtype='object')
(1718, 4)
Date     object
Open    float64
High    float64
Low     float64
dtype: object


Date        Open        High         Low
0  2011-12-15  154.740005  154.949997  151.710007
1  2011-12-16  154.309998  155.369995  153.899994
2  2011-12-19  155.479996  155.860001  154.360001
3  2011-12-20  156.820007  157.429993  156.580002
4  2011-12-21  156.979996  157.529999  156.130005

In [145]:
# Create a subset of the previous subset with only the first 500 rows

gsp_subset_500 = gsp_subset.iloc[:500,:]
print(gsp_subset_500.columns)
print(gsp_subset_500.shape)
print(gsp_subset_500.dtypes)
print(gsp_subset_500.isna().sum()) # Prints the number of missing values per column
gsp_subset_500.head()

Index(['Date', 'Open', 'High', 'Low'], dtype='object')
(500, 4)
Date     object
Open    float64
High    float64
Low     float64
dtype: object
Date    0
Open    0
High    0
Low     0
dtype: int64


Date        Open        High         Low
0  2011-12-15  154.740005  154.949997  151.710007
1  2011-12-16  154.309998  155.369995  153.899994
2  2011-12-19  155.479996  155.860001  154.360001
3  2011-12-20  156.820007  157.429993  156.580002
4  2011-12-21  156.979996  157.529999  156.130005

In [149]:
# Create a subset of the op dataframe with only the 'Date', 'Open', 'High', 'Low' columns, and the first 500 rows

op_subset_500 = op[['Date', 'Open', 'High', 'Low']].iloc[:500]
print(op_subset_500.columns)
print(op_subset_500.shape)
print(op_subset_500.dtypes)
print(op_subset_500.isna().sum()) # Prints the number of missing values per column
op_subset_500.head()

Index(['Date', 'Open', 'High', 'Low'], dtype='object')
(500, 4)
Date     object
Open    float64
High    float64
Low     float64
dtype: object
Date    0
Open    0
High    0
Low     0
dtype: int64


Date    Open    High     Low
0  2011-12-01  100.51  101.17   98.87
1  2011-12-02  100.00  101.56   99.76
2  2011-12-05  101.23  102.44  100.24
3  2011-12-06  100.45  101.42  100.20
4  2011-12-07  101.11  101.94   99.67

No missing values in both subframes.

In [157]:
gsp_subset_500.describe()


Open        High         Low
count  500.000000  500.000000  500.000000
mean   149.033480  149.774340  148.266740
std     16.923214   16.919180   16.871447
min    117.580002  118.180000  116.739998
25%    132.134998  133.144997  131.192501
50%    155.070000  155.610001  154.360001
75%    161.862500  162.577500  161.149994
max    173.199997  174.070007  172.919998

In [103]:
op_subset_500.describe()

Open        High         Low
count  500.000000  500.000000  500.000000
mean    96.196620   97.153780   95.084320
std      6.991871    6.945661    7.003769
min     78.110000   79.680000   77.280000
25%     91.707500   92.627500   90.645000
50%     95.790000   96.640000   94.820000
75%    102.002500  102.962500  100.847500
max    110.280000  112.240000  109.110000

The statistical values for the gold stock prices are

|     | Minimum   | Maximum   | Mean | 75<sup>th</sup> percentile |
|------------|------------|------------|------------|------------|
| Open| 117.58 | 173.20 | 149.03 | 161.86 | 
| High| 118.18 | 174.07 | 149.77 | 162.58 |
| Low | 116.74 | 172.92 | 148.27 | 161.15 |

The statistical values for the oil prices are

|     | Minimum   | Maximum   | Mean | 75<sup>th</sup> percentile |
|------------|------------|------------|------------|------------|
| Open| 78.11 | 110.28 | 96.20 | 102.00 | 
| High| 79.68 | 112.24 | 97.15 | 102.96 |
| Low | 77.28 | 109.11 | 95.08 | 100.85 |

